src/modules/udw/common/select-content-button/select.content.button.component.js

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../../../common/icon/icon';
+import { createCssClassNames } from '../../../common/css-class-names/css.class.names';
+
+const SelectContentButtonComponent = ({ checkCanSelectContent, location, onSelect, onDeselect, isSelected }) => {
+    const [isSelectContentEnabled, setIsSelectContentEnabled] = useState(true);
+    const handleSelect = useCallback(
+        (event) => {
+            event.stopPropagation();
+
+            onSelect(location);
+        },
+        [location, onSelect]
+    );
+    const handleDeselect = useCallback(
+        (event) => {
+            event.stopPropagation();
+
+            onDeselect(location.id);
+        },
+        [location, onDeselect]
+    );
+    const toggleEnabledState = useCallback(
+        (selectContentEnabled) => {
+            if (isSelectContentEnabled === selectContentEnabled) {
+                return;
+            }
+
+            setIsSelectContentEnabled(selectContentEnabled);
+        },
+        [isSelectContentEnabled]
+    );
+
+    useEffect(() => {
+        checkCanSelectContent(location, toggleEnabledState);
```

Comment:
Should we `setIsSelectContentEnabled(false)` before `checkCanSelectContent`?

Reply:
This is not the best idea in the world, because it will cause extra re-renders of a component. Another effect is basing on changes of `isSelectContentEnabled` flag.